bitcoin/src/parse.rs

```diff
@@ -110,6 +110,15 @@ pub(crate) fn hex_u32<S: AsRef<str> + Into<String>>(s: S) -> Result<u32, ParseIn
     })
 }
 
+pub(crate) fn hex_u128(s: &str) -> Result<u128, ParseIntError> {
+    u128::from_str_radix(s, 16).map_err(|error| ParseIntError {
+        input: s.into(),
+        bits: u8::try_from(core::mem::size_of::<u128>() * 8).expect("max is 128 bits for u128"),
```

Comment:
In 841a185bb66c9b2ac6587ac01ef87cf19651562e:

I find this call to `core::mem::size_of` confusing. I get that it's from generic code but it looks like it's designed to break our serialization in the case that somehow `u128`'s in-memory size ever changes. I think it'd be way clearer to just type 128.

Reply:
Face palm, cargo cult programming combined with tunnel vision - I didn't even see this yesterday when I fixed `is_signed`.